File: src/walkway_navigation/src/map_monitor.cpp
#include "rclcpp/rclcpp.hpp"
#include "nav_msgs/msg/occupancy_grid.hpp"

class MapMonitor : public rclcpp::Node {

    public:
    MapMonitor() : Node('map_monitor') {
        subscription_ = this->create_subscription<nav_mags::msg::OccupancyGrid>(
            "/map", 10, std::bind(&MapMonitor::map_callback, this, std::placeholders::_1)
        );
    }

    private:
    void map_callback(const nav_msgs::msg::OccupancyGrid::SharedPtr msg) const {
        RCLCPP_INFO(this->get_logger(),
            "Received map! Size: %d x %d, Resolution: %f m/cell",
            msg->info.width, msg->info.height, msg->info.resolution);
    }

    rclcpp::Subscription<nav_msgs::msg::OccupancyGrid>::SharedPtr subscription_:
};

int main(int argc, char * argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<MapMonitor>());
    rclcpp::shutdown();
    return 0;
}

File: src/walkway_navigation/scripts/map_monitor.py
#!/usr/bin/env python3 

import rclpy
from rclpy.node import Node
from nav_msgs.msg import OccupancyGrid

class MapMonitor(Node):
    def __init__(self):
        super().__init__('map_monitor')
        self.subscription = self.create_subscription(
            OccupancyGrid, '/map', self.map_callback,10
        )

    def map_callback(self, msg):
        self.get_logger().info(
            f'Received map! Size: {msg.info.width}x{msg.info.height}, '
            f'Resolution: {msg.info.resolution} m/cell'
        )

def main(args=None):
    rclpy.init(args=args)
    node = MapMonitor()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()